src/libraries/System.Net.Security/src/System/Net/Security/SslConnectionInfo.Unix.cs

```diff
@@ -12,14 +17,2710 @@ internal partial struct SslConnectionInfo
         private void MapCipherSuite(TlsCipherSuite cipherSuite)
         {
             TlsCipherSuite = cipherSuite;
-
-            TlsCipherSuiteData data = TlsCipherSuiteData.GetCipherSuiteData(cipherSuite);
-            KeyExchangeAlg = (int)data.KeyExchangeAlgorithm;
             KeyExchKeySize = 0;
-            DataCipherAlg = (int)data.CipherAlgorithm;
-            DataKeySize = data.CipherAlgorithmStrength;
-            DataHashAlg = (int)data.MACAlgorithm;
-            DataHashKeySize = data.MACAlgorithmStrength;
+
+            switch (cipherSuite)
```

Comment:
What does Roslyn compile this big switch into? Is it creating some side hashtable? We may be able to do better by creating the hashtable ourselves.

Reply:
Seems like Roslyn is compiling it to switch-case with minor restructuring:
https://sharplab.io/#gist:522bb28287b7c9fcbf6fd65f16e93315